packages/data-controls/src/index.js

```diff
@@ -31,98 +31,22 @@ export const apiFetch = ( request ) => {
- *
- * @param {string} storeKey      The key for the store the selector belongs to
- * @param {string} selectorName  The name of the selector
- * @param {Array}  args          Arguments for the selector.
- *
- * @example
- * ```js
- * import { select } from '@wordpress/data-controls';
- *
- * // Action generator using select
- * export function* myAction() {
- * 	const isSidebarOpened = yield select( 'core/edit-post', 'isEditorSideBarOpened' );
- * 	// do stuff with the result from the select.
- * }
- * ```
- *
- * @return {Object} The control descriptor.
+ * Control for resolving a selector in a registered data store.
+ * Alias for the `resolveSelect` built-in control in the `@wordpress/data` package.
  */
-export function select( storeKey, selectorName, ...args ) {
-	return {
-		type: 'SELECT',
-		storeKey,
-		selectorName,
-		args,
-	};
-}
+export const select = dataControls.resolveSelect;
 
 /**
- * Dispatches a control action for triggering a synchronous registry select.
- *
- * Note: This control synchronously returns the current selector value, triggering the
- * resolution, but not waiting for it.
- *
- * @param {string} storeKey     The key for the store the selector belongs to.
- * @param {string} selectorName The name of the selector.
- * @param {Array}  args         Arguments for the selector.
- *
- * @example
- * ```js
- * import { syncSelect } from '@wordpress/data-controls';
- *
- * // Action generator using `syncSelect`.
- * export function* myAction() {
- * 	const isEditorSideBarOpened = yield syncSelect( 'core/edit-post', 'isEditorSideBarOpened' );
- * 	// Do stuff with the result from the `syncSelect`.
- * }
- * ```
- *
- * @return {Object} The control descriptor.
+ * Control for calling a selector in a registered data store.
+ * Alias for the `select` built-in control in the `@wordpress/data` package.
  */
-export function syncSelect( storeKey, selectorName, ...args ) {
-	return {
-		type: 'SYNC_SELECT',
-		storeKey,
-		selectorName,
-		args,
-	};
-}
+export const syncSelect = dataControls.select;
 
 /**
- * Dispatches a control action for triggering a registry dispatch.
- *
- * @param {string} storeKey    The key for the store the action belongs to
- * @param {string} actionName  The name of the action to dispatch
- * @param {Array}  args        Arguments for the dispatch action.
- *
- * @example
- * ```js
- * import { dispatch } from '@wordpress/data-controls';
- *
- * // Action generator using dispatch
- * export function* myAction() {
- * 	yield dispatch( 'core/edit-post', 'togglePublishSidebar' );
- * 	// do some other things.
- * }
- * ```
- *
- * @return {Object}  The control descriptor.
+ * Control for dispatching an action in a registered data store.
+ * Alias for the `dispatch` control in the `@wordpress/data` package.
  */
-export function dispatch( storeKey, actionName, ...args ) {
-	return {
-		type: 'DISPATCH',
-		storeKey,
-		actionName,
-		args,
-	};
-}
+export const dispatch = dataControls.dispatch;
```

Comment:
do you think we should deprecate these?

Reply:
Yes, deprecation would discourage folks from adding new usages of `data-controls`. I guess the `data-controls` wrappers should use the `@wordpress/deprecated` helper?

Reply:
@youknowriad I updated the `data-controls` wrappers to warn about deprecation using the `@wordpress/deprecated` package, in https://github.com/WordPress/gutenberg/pull/25362/commits/51f7fc52d7bcbe052a84b2352a02ea326c504f70.

But now I have problems with unit tests that I don't know how to solve: every unit tests uses the `jest-console` plugin that checks that the code didn't produce any console messages. But the `deprecated()` function logs the deprecation messages with `console.warn()`, so many unit tests for code that still uses the deprecated controls now fail.

Is there any established way how to get out of this?

Reply:
Yes, we have some asserters to aknowledge that warnings will be thrown. `expect( console ).toHaveWarned();`

That said, when we introduce deprecations, we do it after refactoring the Core usage of this API. Do we still have existing places where these APIs are used on Gutenberg?

Reply:
> Do we still have existing places where these APIs are used on Gutenberg?

Yes, the `data-controls` are still used in many stores. It will take some 5+ PRs to migrate them one by one.

At this moment, it seems to me the best option is to mark the `data-controls` as deprecated only after that migration is finished. It shouldn't take long after this PR is merged.

Reply:
> it seems to me the best option is to mark the data-controls as deprecated only after that migration is finished. It shouldn't take long after this PR is merged.

👍